modAionImpl/src/org/aion/zero/impl/sync/ReceiptsRetrievalVerifier.java

```diff
@@ -0,0 +1,124 @@
+package org.aion.zero.impl.sync;
+
+import org.aion.base.util.ByteUtil;
+import org.aion.log.AionLoggerFactory;
+import org.aion.log.LogEnum;
+import org.aion.mcf.db.TransactionStore;
+import org.aion.p2p.IP2pMgr;
+import org.aion.zero.impl.core.IAionBlockchain;
+import org.aion.zero.impl.db.AionRepositoryImpl;
+import org.aion.zero.impl.sync.msg.ReqTxReceipts;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.impl.types.AionTxInfo;
+import org.aion.zero.types.AionTransaction;
+import org.slf4j.Logger;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Verify receipts requests process by:
+ *  1) comparing the list of tx receipts we've requested against the list of tx receipts we've received.
+ *  2) checking that the received receipt matches that of the receipt we would have made if we had computed it locally
```

Comment:
Formatting will make this harder to read, but you can use html tags to create an ordered list.